go-controller/pkg/util/util.go

```diff
@@ -139,3 +139,37 @@ func IsAnnotationNotSetError(err error) bool {
 	_, ok := err.(annotationNotSetError)
 	return ok
 }
+
+// CalculateHostSubnetsForClusterEntry calculates the host subnets
+// available in a CIDR entry
+func CalculateHostSubnetsForClusterEntry(cidrEntry config.CIDRNetworkEntry,
+	v4HostSubnetCount, v6HostSubnetCount *float64) {
+	prefixLength, _ := cidrEntry.CIDR.Mask.Size()
+	var one uint64 = 1
+	if prefixLength > cidrEntry.HostSubnetLength {
+		klog.Warningf("Invalid cidr entry: %+v found while calculating subnet count",
+			cidrEntry)
+		return
+	}
+	if !utilnet.IsIPv6CIDR(cidrEntry.CIDR) {
+		*v4HostSubnetCount = *v4HostSubnetCount + float64(one<<(cidrEntry.HostSubnetLength-prefixLength))
+	} else {
+		*v6HostSubnetCount = *v6HostSubnetCount + float64(one<<(cidrEntry.HostSubnetLength-prefixLength))
+
+	}
+}
+
+// UpdateUsedHostSubnetsCount increments the v4/v6 host subnets count based on
+// the subnet being visited
+func UpdateUsedHostSubnetsCount(subnet *net.IPNet,
```

Comment:
why don' t you use Inc() and Dec()?
https://godoc.org/github.com/prometheus/client_golang/prometheus#Gauge

Reply:
Using the Inc() and Dec() counters will need some code duplication at several places we will call UpdateUsedHostSubnetsCount(). It's quite doable, I just thought, hiding away the details of setting the metric i.e. whether to Inc() and Dec() and whether if its a v4 or v6 subnet etc. could be hidden from the users and we could only call Set() once across all the subnets for a given host.